File: December 2024 Leetcode Solution/December-06.cpp
class Solution {
 public:
  int maxCount(vector<int>& banned, int n, int maxSum) {
    sort(banned.begin(), banned.end());  
    int ans = 0, sum = 0, bannedIndex = 0;

    for (int i = 1; i <= n; ++i) {
      while (bannedIndex < banned.size() && banned[bannedIndex] < i)
        ++bannedIndex;
      
      if (bannedIndex < banned.size() && banned[bannedIndex] == i)
        continue;

      if (sum + i > maxSum)
        break;

      sum += i;
      ++ans;
    }

    return ans;
  }
};

File: December 2024 Leetcode Solution/December-06.py
class Solution:
    def maxCount(self, banned: list[int], n: int, maxSum: int) -> int:
        banned_set = set(banned)  
        ans = 0
        total_sum = 0

        for i in range(1, n + 1):
            if i in banned_set:  
                continue

            if total_sum + i > maxSum:
                break

            total_sum += i
            ans += 1

        return ans
